Moved platform communication from form to JSON

cosminbosutar opened the pull request.

cosminbosutar committed:
Overwrite version 0.0.1 with 1.0.1 changes

Overwrite version 0.0.1 with 1.0.1 changes that have been extracted from the woocommerce marketplace.

cosminbosutar committed:
Moved platform communication from form to JSON

Moved platform communication from classic form with multiple inputs, to form containing JSON encoded data.

cosminbosutar committed:
Update twispay.zip

Update twispay.zip with the latest changes

dragos-ursu reviewed (changes requested):
Please also add content in the `readme.txt` with informations about the release (including the changes made).

dragos-ursu commented on Twispay WooCommerce Plugin/twispay/includes/twispay-processor.php:
```diff
@@ -21,236 +23,151 @@
-        
-        // Get configuration from database
+
+        error_log( print_r( $data, true ) );
```
Please remove this line of code (with the `error_log`).

dragos-ursu commented on Twispay WooCommerce Plugin/twispay/includes/twispay-processor.php:
```diff
@@ -21,236 +23,151 @@
-        // Site ID and Private Key
+        $configuration = $wpdb->get_row( "SELECT * FROM " . $wpdb->prefix . "twispay_tw_configuration" );
+
+        error_log( print_r( $data, true ) );
```
Please remove this line of code (with the `error_log`).

dragos-ursu commented on Twispay WooCommerce Plugin/twispay/includes/twispay-processor.php:
```diff
@@ -21,236 +23,151 @@
+                    , 'zipCode' => ($data['billing']['postcode']) ? ($data['billing']['postcode']) : ($data['shipping']['postcode'])
+                    /* , 'phone' => $data['billing']['phone'] */
+                    , 'email' => $data['billing']['email']
+                    , 'tags' => []
```
We should comment this line as `tags` is not a required field.

cosminbosutar replied:
Done

dragos-ursu commented on Twispay WooCommerce Plugin/twispay/includes/twispay-processor.php:
```diff
@@ -21,236 +23,151 @@
+                                  , 'amount' => $data['total']
+                                  , 'currency' => $data['currency']
+                                  , 'items' => $items
+                                  , 'tags' => []
```
We should comment this line as `tags` is not a required field.

cosminbosutar replied:
Done

dragos-ursu commented on Twispay WooCommerce Plugin/twispay/includes/twispay-processor.php:
```diff
@@ -21,236 +23,151 @@
+                                  /*                      , 'travelAgencyName' => ''] */
+                                  ]
+                    , 'cardTransactionMode' => 'authAndCapture'
+                    , 'cardId' => 0
```
We should comment this line as `cardId` is not a required field (and setting it to `0` value means it will not be used).

cosminbosutar replied:
Done

cosminbosutar committed:
Pull request review code changes

cosminbosutar committed:
Code indentation

cosminbosutar committed:
Renamed helper classes, added phone to JSON and changed backUrl construction mode

Renamed helper classes.
Added the phone added by the client in the billing info to the server notification JSON.
Made the construction of the "backUrl" dependant of the permalinks settings

cosminbosutar committed:
Update readme.txt

Updated readme after review

cosminbosutar committed:
Update twispay-processor.php

Updated the way the timestamp is added.
